File: src/typescript/examples/basic_usage.ts
import { GeminiClient, EventType } from '../src/index.js';

async function main(): Promise<void> {
  console.log('GeminiSDK TypeScript - Basic Usage Example\n');

  // Create client with default options
  const client = new GeminiClient();

  try {
    // Start the client (authenticates with Gemini CLI credentials)
    console.log('Starting client...');
    await client.start();
    console.log('Client started and authenticated!\n');

    // List available models
    console.log('Available models:');
    const models = await client.listModels();
    for (const model of models) {
      console.log(`  - ${model.name} (${model.id})`);
    }
    console.log();

    // Create a session
    console.log('Creating session...');
    const session = await client.createSession({
      model: 'gemini-2.5-flash',
      streaming: false, // Non-streaming for simplicity
      systemMessage: 'You are a helpful assistant.',
    });
    console.log(`Session created: ${session.sessionId}\n`);

    // Send a message and wait for response
    console.log('Sending message...');
    const response = await session.sendAndWait({
      prompt: 'What are three interesting facts about the TypeScript programming language?',
    });

    console.log('Response received:');
    console.log(`Event type: ${response.type}`);
    const data = response.data as Record<string, unknown>;
    if (data?.content) {
      console.log(`Content: ${data.content}`);
    }

    // Clean up
    console.log('\nClosing client...');
    await client.close();
    console.log('Done!');
  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

main();

File: src/rust/examples/basic_usage.rs
//! Basic usage example for GeminiSDK Rust

use geminisdk::{GeminiClient, GeminiClientOptions, MessageOptions, SessionConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize logging
    env_logger::init();

    println!("GeminiSDK Rust - Basic Usage Example\n");

    // Create client with default options
    let client = GeminiClient::new(GeminiClientOptions::default());

    // Start the client (authenticates with Gemini CLI credentials)
    println!("Starting client...");
    client.start().await?;
    println!("Client started and authenticated!\n");

    // List available models
    println!("Available models:");
    for model in client.list_models().await {
        println!("  - {} ({})", model.name, model.id);
    }
    println!();

    // Create a session
    println!("Creating session...");
    let session = client
        .create_session(SessionConfig {
            model: Some("gemini-2.5-flash".to_string()),
            streaming: Some(false), // Non-streaming for simplicity
            system_message: Some("You are a helpful assistant.".to_string()),
            ..Default::default()
        })
        .await?;

    println!("Session created: {}\n", session.session_id());

    // Send a message and wait for response
    println!("Sending message...");
    let response = session
        .send_and_wait(MessageOptions {
            prompt: "What are three interesting facts about the Rust programming language?"
                .to_string(),
            attachments: None,
            context: None,
        })
        .await?;

    println!("Response received:");
    println!("Event type: {:?}", response.event_type);
    if let Some(content) = response.data.get("content") {
        println!("Content: {}", content);
    }

    // Clean up
    println!("\nClosing client...");
    client.close().await?;
    println!("Done!");

    Ok(())
}
